packages/lesswrong/lib/ultraFeed/UltraFeedRepo.ts

```diff
@@ -0,0 +1,413 @@
+           
+           if (!scenario) {
+             // Try to detect special posts first, then try recommId
+             const aboutPostId = aboutPostIdSetting.get();
+             if (aboutPostId && item.post._id === aboutPostId && idx === 0) {
+               scenario = 'welcome-post';
+             } else if (item.curated) {
+               scenario = 'curated';
+             } else if (item.stickied || item.post.sticky) { 
+               scenario = 'stickied';
+             } else if (item.recommId) {
+               if (item.recommId.includes('forum-classic')) {
+                 scenario = 'hacker-news';
+               } else if (item.recommId.includes('recombee-lesswrong-custom')) {
+                 scenario = 'recombee-lesswrong-custom';
+               }
+             } else {
+               // Default fallback: Use empty sources array and warn
+               scenario = undefined; // Explicitly set scenario to undefined for clarity
+             }
+           }
+           
+           let sources: FeedItemSourceType[];
+           if (scenario) {
+             sources = [scenario as FeedItemSourceType];
+           } else {
+             sources = [];
+             // eslint-disable-next-line no-console
+             console.warn(`UltraFeedRepo: Could not determine specific source scenario for post ${item.post._id}. Assigning empty sources.`);
+           }
+           
+           const recommInfo = (item.recommId && item.generatedAt) ? {
+             recommId: item.recommId,
+             scenario: scenario || 'unknown',
+             generatedAt: item.generatedAt,
+           } : undefined;
+
+           return {
+             post: item.post,
+             postMetaInfo: {
+               sources: sources,
+               displayStatus: 'expanded',
+               recommInfo: recommInfo,
+             },
+           };
+        }).filter((p): p is FeedFullPost => p !== null);
+
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error("Error calling getHybridRecommendationsForUser:", error);
+      }
+    }
+    
+    return displayPosts; 
+  }
+
+  //TODO: add comments on threads you've partificipated in/interacted with, especially replies to you
+  //TODO: figure out exact threshold / date window (make parameter?), perhaps make a window that's adjusted based on user's visit frequency
+
+  async getCommentsForFeed(context: ResolverContext, maxTotalComments = 1000): Promise<FeedCommentFromDb[]> {
+    const db = this.getRawDb();
+    const userId = context.userId;
+    const initialCandidateLimit = 100; // Limit for the first pass
+
+    const UNIVERSAL_COMMENT_FILTER_CLAUSE = `
+      c.deleted IS NOT TRUE
+      AND c.retracted IS NOT TRUE
+      AND c."authorIsUnreviewed" IS NOT TRUE
+      AND c."postId" IS NOT NULL
+    `;
+
+    const CANDIDATE_DATE_FILTER_CLAUSE = `
+      c."postedAt" > NOW() - INTERVAL '${COMMENT_LOOKBACK_WINDOW}'
```

Comment:
If `COMMENT_LOOKBACK_WINDOW` is only used in one place I'd prefer it to just be inlined.